contrib/aws-efa/aws_test.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -exE -o pipefail
+
+# Get test type from argument, default to cpp
+TEST_TYPE=${1:-"cpp"}
+
+# Set AWS container image (can be overridden via environment)
+export CONTAINER_IMAGE=${CONTAINER_IMAGE:-"nvcr.io/nvidia/pytorch:25.02-py3"}
+
+# Set Git checkout command based on GITHUB_REF
+if [ -z "$GITHUB_REF" ]; then   # manual run
+    echo "Error: GITHUB_REF environment variable must be set"
+    echo "For a branch, use: export GITHUB_REF=\"main\""
+    echo "For a PR, use: export GITHUB_REF=\"refs/pull/187/head\""
+    exit 1
+fi
+
+case "$GITHUB_REF" in
+    refs/pull/*)
+        export GIT_CHECKOUT_CMD="git fetch origin ${GITHUB_REF} && git checkout FETCH_HEAD"
+        ;;
+    *)
+        export GIT_CHECKOUT_CMD="git checkout ${GITHUB_REF}"
+        ;;
+esac
+
+# Construct command to run in AWS
+setup_cmd="set -x && \
+    git clone ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY} && \
+    cd nixl && \
+    ${GIT_CHECKOUT_CMD}"
+build_cmd=".gitlab/build.sh \${NIXL_INSTALL_DIR} \${UCX_INSTALL_DIR}"
+test_script="test_${TEST_TYPE}"
```

Comment:
maybe we can pass the test script path as argument? so we can run as many scripts we want

Reply:
Done. Now we can add multiple scripts per test suite.
https://github.com/ai-dynamo/nixl/pull/212/files#diff-0d0db0974df99ada87d9b3aa237c7bc6aebd8ebeb696841cc362e73cf78f4783R24